src/sdk/wire_bytes.rs

```diff
@@ -27,41 +28,62 @@ pub fn encode<T: Serialize>(payload: &T) -> TofnResult<BytesVec> {
     })
 }
 
+/// Serialize a value using bincode and log errors
 pub fn serialize<T: ?Sized>(value: &T) -> TofnResult<BytesVec>
 where
     T: serde::Serialize,
 {
-    match bincode::serialize(value) {
-        Ok(bytes) => Ok(bytes),
-        Err(err) => {
-            error!("serialization failure: {}", err.to_string());
-            Err(TofnFatal)
-        }
-    }
+    // Create serialization options for bincode.
```

Comment:
@milapsheth beware: this function is also used for share recovery, which encodes much more data than any protocol message.  Moreover, the byte length of recovery data is non-constant (scales with number of shares).  Thus, any limits you set on message size applies also to share recovery, so you could run into problems.  That's why I put the length check in `Round::msg_in` instead.  See https://github.com/axelarnetwork/tofn/issues/169#issuecomment-931689985

Reply:
Oh, yes, good point!. Well, doesn't make sense to bound the `encode/decode` with a tight length then. Maybe some sanity limit like `1 MB` could still be used.

Reply:
I swear I posted this comment https://github.com/axelarnetwork/tofn/pull/174#discussion_r720895411 before I read your comment above. 😛 Glad to see we're both on the same page.

Reply:
Haha, awesome!